### PR TITLE
nautilus: fix TestExports failure

### PR DESCRIPTION
**Pull request by ajarr**
Fixes: https://tracker.ceph.com/issues/45967
Fixes: https://tracker.ceph.com/issues/45974

### CODE REVIEW

**Comment by ajarr** [comment]
@sidharthanup , can you please test this in teuthology with priority "10" and use filter option to run only the test_exports test? This way we can try merging this PR today.

**Comment by sidharthanup** [comment]
Ack. Scheduled a run here: http://pulpito.ceph.com/sidharthanup-2020-06-11_08:40:17-multimds-nautilus-distro-basic-smithi/

File: qa/tasks/cephfs/test_exports.py [hunk 1]
```diff
@@ -131,12 +131,12 @@ def test_export_pin_getfattr(self):
             p = self.mount_a.client_remote.sh('uname -r', wait=True)
             dir_pin = self.mount_a.getfattr("1", "ceph.dir.pin")
             log.debug("mount.getfattr('1','ceph.dir.pin'): %s " % dir_pin)
-            if str(p.stdout.getvalue()) < "5" and not(dir_pin):
+            if str(p) < "5" and not(dir_pin):
                 self.skipTest("Kernel does not support getting the extended attribute ceph.dir.pin")
-        self.assertTrue(self.mount_a.getfattr("1", "ceph.dir.pin") == "1")
-        self.assertTrue(self.mount_a.getfattr("1/2", "ceph.dir.pin") == "0")
+        self.assertEqual(self.mount_a.getfattr("1", "ceph.dir.pin"), "1")
```
Comment (tchaikov) [inline qa/tasks/cephfs/test_exports.py:136]: i am confused. in this change you backport a change which breaks the test, and then create a commit which fixes it.

Comment (ajarr) [reply qa/tasks/cephfs/test_exports.py:136]: @tchaikov so you suggest that I alter the cherry-pick commit itself (IOW merge the commits)? and then mention in the cherry-pick commit message that I'd to make some changes to make it work with nautilus?

Comment (sidharthanup) [reply qa/tasks/cephfs/test_exports.py:136]: @tchaikov Only the `"1" -> b'1'` change is what breaks it. The rest of the changes in that commit are necessary. Hence the subsequent commit to fix just that.

Comment (tchaikov) [reply qa/tasks/cephfs/test_exports.py:136]: no, i think what you need is 

1. backport b98f3e388b6a38e91c2ee064466ae8f130a5bcda as this PR already have it
1. backport 3c1330795add81126424e9e0a49cc07881fbdf8f

Comment (ajarr) [reply qa/tasks/cephfs/test_exports.py]: Ah! Got it! :-) Thanks a lot @tchaikov

**Comment by ajarr** [comment]
@tchaikov @sidharthanup PTAL

**Comment by sidharthanup** [comment]
LGTM. Scheduled a new run: http://pulpito.ceph.com/sidharthanup-2020-06-11_10:25:02-multimds-nautilus-distro-basic-smithi/

**Comment by ajarr** [comment]
Tests passed. Good to merge?

**Pull request merged**